pkg/steps/output_image_tag.go

```diff
@@ -54,16 +56,16 @@ func (s *outputImageTagStep) Run(ctx context.Context, dry bool) error {
 		return nil
 	}
 
-	// TODO: this step is "force update" today, but that behavior should be optional in the future
-	//   since other steps are "idempotent". However, the override case supports promotion of machine-os-content
-	//   and will be fixed as part of that.
-	if err := s.istClient.ImageStreamTags(toNamespace).Delete(ist.Name, nil); err != nil && !errors.IsNotFound(err) {
-		return fmt.Errorf("could not remove output imagestreamtag: %v", err)
-	}
-
-	// Create if not exists, if it already exists, then we have nothing to do.
-	if _, err := s.istClient.ImageStreamTags(toNamespace).Create(ist); err != nil && !errors.IsAlreadyExists(err) {
-		return fmt.Errorf("could not create output imagestreamtag: %v", err)
+	// ensure that the image stream tag points to the correct input, retry
+	// on conflict, and do nothing if another user creates before us
+	if err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		_, err := s.istClient.ImageStreamTags(toNamespace).Update(ist)
+		if errors.IsNotFound(err) {
```

Comment:
How often the image already exists? Do you think that it's better first to try to create the image and if its already exist then update?

Reply:
the image will almost always exist (we create stable based on the release tags, then we modify it)